Core/src/org/sleuthkit/autopsy/coreutils/textutils/EncodingUtils.java

```diff
@@ -87,12 +87,27 @@ public static Charset getEncoding(AbstractFile file) throws TskCoreException, IO
         try (InputStream stream = new BufferedInputStream(new ReadContentInputStream(file))) {
             CharsetDetector detector = new CharsetDetector();
             detector.setText(stream);
-            CharsetMatch tikaResult = detector.detect();
-            if (tikaResult != null && tikaResult.getConfidence() >= MIN_CHARSETDETECT_MATCH_CONFIDENCE) {
-                String tikaCharSet = tikaResult.getName();
-                //Check if the nio package has support for the charset determined by Tika.
-                if(Charset.isSupported(tikaCharSet)) {
-                    return Charset.forName(tikaCharSet);
+            
+            CharsetMatch[] tikaResults = detector.detectAll();
+            // Get all guesses by Tika. These CharsetMatch's are ordered
+            // by descending confidence (largest first).
+            if (tikaResults.length > 0) {
+                CharsetMatch topPick = tikaResults[0];
+                
+                if (topPick.getName().equalsIgnoreCase("IBM500") && tikaResults.length > 1) {
+                    // Legacy encoding, let's discard this one in favor
+                    // of the second pick. Tika has some problems with 
+                    // mistakenly identifying text as IBM500. See JIRA-6600 
```

Comment:
We should put the Tika story in here too.

Reply:
Done :)